cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/image/AzureImageSetupService.java

```diff
@@ -49,6 +49,9 @@ public ImageStatusResult checkImageStatus(AuthenticatedContext ac, CloudStack st
         String imageStorageName = armStorage.getImageStorageName(acv, ac.getCloudContext(), stack);
         try {
             CopyState copyState = client.getCopyStatus(imageResourceGroupName, imageStorageName, IMAGES_CONTAINER, image.getImageName());
+            if (copyState == null && storageContainsImage(client, imageResourceGroupName, imageStorageName, image.getImageName())) {
+                return new ImageStatusResult(ImageStatus.CREATE_FINISHED, ImageStatusResult.COMPLETED);
+            }
             if (CopyStatus.SUCCESS.equals(copyState.getStatus())) {
```

Comment:
This can still throw NPE when the storage does not contain the image?

Reply:
You're right. It's fixed now.

Reply:
Thanks.